src/main/scala/scala.camp/routes/UserRoutes.scala

```diff
@@ -0,0 +1,71 @@
+package scala.camp.routes
+
+import akka.http.scaladsl.model.StatusCodes.NotFound
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server._
+
+import scala.camp.model.{JsonSupport, User}
+import scala.camp.repository.UserRepositoryImpl
+import scala.camp.utils.{Retry, Validator}
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+trait UserRoutes extends UserRepositoryImpl with JsonSupport with Retry {
+
+  private val emailRegex =
+    """^[a-zA-Z0-9\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$""".r
+
+  def checkEmail(e: String): Boolean = e match {
+    case null                                          => false
+    case e if e.trim.isEmpty                           => false
+    case e if emailRegex.findFirstMatchIn(e).isDefined => true
+    case _                                             => false
+  }
+
+  def emailValidator = new Validator[String] {
+    override def validate(value: String): Either[String, String] =
+      if (checkEmail(value)) Right(value) else Left(s"Email '$value' is Not valid!")
+  }
+
+  def retryUserPost(user: User): Future[Int] =
+    retryf[Int](() => registerUser(user), id => id != 0, List(1.seconds, 2.seconds, 3.seconds))
+
+  def retryUserGet(id: Int): Future[Option[User]] =
+    retryf[Option[User]](() => getById(id),
+                         user => user.isDefined,
```

Comment:
if there is no user with such id in the database what is the point to retry that?